sb_vision/calibration/training_example.py

```diff
@@ -1,10 +1,36 @@
 """Definitions for training examples."""
 
-import collections
+from ..camera import FileCamera
+from ..vision import Vision
 
 
-TrainingExample = collections.namedtuple('TrainingExample', (
-    'image_file',
-    'z_distance',
-    'x_offset_right',
-))
+class TrainingExample(object):
+    """A single example (image/pose pair) for model training."""
+
+    def __init__(self, *, image_file, z_distance, x_offset_right=0.0):
+        """Construct from image file and distance metrics."""
+        self.z_distance = z_distance
+        self.x_offset_right = x_offset_right
+        self.image_file = image_file
+
+        self._load()
+
+    def _load(self):
+        """Load and recognise the single marker from this image."""
+
+        pseudo_camera = FileCamera(self.image_file, focal_length=None)
+        vision = Vision(pseudo_camera, (0.01, 0.01))
+
+        (single_token,) = vision.snapshot()
```

Comment:
That has different semantics: `(foo,)` unpacks a single-element iterable whereas `foo, _` unpacks a two-element tuple ignoring the second.

I'm not averse to writing this `single_token, =` and dropping the parens if you'd prefer, I tend to include the parens to make it more visually distinct from `single_token =`.